spec/spec.go

```diff
@@ -24,16 +28,52 @@ func isSwagger(spec []byte) bool {
 	return header.Swagger != ""
 }
 
-func ParseFromFile(path string) (*openapi3.T, error) {
-	contents, err := os.ReadFile(path)
+// Parse is the entrypoint for the spec package
+// Accepts a path that should be parseable into a resource locater
+// i.e. a URL or relative file path
+func Parse(path string) (*openapi3.T, error) {
+	u, err := url.Parse(path)
 	if err != nil {
-		return nil, fmt.Errorf("failed to read spec file: %w", err)
+		return nil, fmt.Errorf("could not parse resource path: %w", err)
 	}
 
-	return Parse(contents)
+	if isURLRelative := u.Host == ""; isURLRelative {
+		return parseFromFile(path)
+	}
+
+	return parseFromURL(u)
 }
 
-func Parse(spec []byte) (*openapi3.T, error) {
+func parseFromURL(u *url.URL) (*openapi3.T, error) {
+	resp, err := http.Get(u.String())
```

Comment:
doesn't the kin-openapi library have a loader that loads from a URL? if so that might be a better option as it should support the resolving of external path refs (see https://swagger.io/specification/#path-item-object)

Reply:
Ah yes it does! Good shout, I will incorporate that in then!